consensus/istanbul/backend/backend.go

```diff
@@ -229,13 +229,28 @@ func (sb *backend) checkInSubList(prevHash common.Hash, valSet istanbul.Validato
 }
 
 // getTargetReceivers returns a map of nodes which need to receive a message
-func (sb *backend) getTargetReceivers(valSet istanbul.ValidatorSet) map[common.Address]bool {
+func (sb *backend) getTargetReceivers(prevHash common.Hash, valSet istanbul.ValidatorSet) map[common.Address]bool {
 	targets := make(map[common.Address]bool)
 
-	for _, val := range valSet.List() {
-		if val.Address() != sb.Address() {
-			targets[val.Address()] = true
+	cv := sb.currentView.Load().(*istanbul.View)
+	view := &istanbul.View{
+		Round:    big.NewInt(cv.Round.Int64()),
+		Sequence: big.NewInt(cv.Sequence.Int64()),
+	}
+
+	proposer := valSet.GetProposer()
+	for i := 0; i < 2; i++ {
+		vals := valSet.SubListWithProposer(prevHash, proposer.Address(), view)
+		for _, val := range vals {
+			if val.Address() != sb.Address() {
+				targets[val.Address()] = true
+			}
+		}
+		// Don't change a proposer for the next round if there is only one node.
+		if len(vals) > 1 {
+			proposer = vals[1]
```

Comment:
Is there no function to get next validator?

Reply:
@KimKyungup As far as I know, the answer is no. But the address of the next proposer is stored in the second slot of the committee list.